modules/core/include/opencv2/core/check.hpp

```diff
@@ -0,0 +1,129 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_CORE_CHECK_HPP
+#define OPENCV_CORE_CHECK_HPP
+
+#include <opencv2/core/base.hpp>
+
+namespace cv {
+
+//! @cond IGNORED
+namespace detail {
+
+/** Returns string of cv::Mat depth value: CV_8U -> "CV_8U" or "CV_InvalidDepth" */
+CV_EXPORTS const char* getMatDepth(int depth);
+
+/** Returns string of cv::Mat depth value: CV_8U -> "CV_8U" or NULL */
+CV_EXPORTS const char* getMatDepth_(int depth);
+
+/** Returns string of cv::Mat depth value: CV_8UC3 -> "CV_8UC3" or "CV_InvalidType" */
+CV_EXPORTS const cv::String getMatType(int type);
```

Comment:
drop `cv::`

rename `getMatType` to `typeToString()`?

same question about 2 functions